bluesky_widgets/qt/search_input.py

```diff
@@ -212,3 +215,18 @@ def uncheck_radiobuttons(self):
         self.today_widget.setChecked(False)
         self.hour_widget.setChecked(False)
         self.radio_button_group.setExclusive(True)
+
+
+@contextlib.contextmanager
+def _blocked(qobject):
+    "Block signals from this object inside the context."
+    qobject.blockSignals(True)
+    yield
+    qobject.blockSignals(False)
+
+
+# We need some concrete datetime to show in the "Since" datetime picker when
+# the "All" button is checked. It should be some time old enough that it isn't
+# likely to leave out wanted data. We cheekily choose this birthday as that
+# time.
+ADA_LOVELACE_BIRTHDAY = datetime(1815, 12, 10)
```

Comment:
I get 1815-12-09-23:52 as all since display. Which I find a bit weird, as it is not n hours different depending on the time zone, but feels a bit random, since it's eg 4:39 pm right now.

Reply:
Weird. The conversion from `datetime.datetime` to `QDateTime` seems to do something funny:

```py
In [5]: ADA_LOVELACE_BIRTHDAY                                                                                                                                                                 
Out[5]: datetime.datetime(1815, 12, 10, 0, 0)

In [6]: as_qdatetime(ADA_LOVELACE_BIRTHDAY)                                                                                                                                                   
Out[6]: PyQt5.QtCore.QDateTime(1815, 12, 9, 23, 56, 2)
```

Investigating....

Reply:
I finally figured this out and added some additional tests to ensure specifically that we convert from `datetime` to `QDateTime` successfully.